ui/pages/send/gas-display/gas-display.component.js

```diff
@@ -0,0 +1,371 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { PRIMARY, SECONDARY } from '../../../helpers/constants/common';
+
+import {
+  addHexes,
+  hexWEIToDecGWEI,
+} from '../../../helpers/utils/conversions.util';
+import UserPreferencedCurrencyDisplay from '../../../components/app/user-preferenced-currency-display/user-preferenced-currency-display.component';
+import GasTiming from '../../../components/app/gas-timing/gas-timing.component';
+import InfoTooltip from '../../../components/ui/info-tooltip/info-tooltip';
+import LedgerInstructionField from '../../../components/app/ledger-instruction-field/ledger-instruction-field';
+import TextField from '../../../components/ui/text-field';
+import Typography from '../../../components/ui/typography/typography';
+import Button from '../../../components/ui/button';
+import { TYPOGRAPHY } from '../../../helpers/constants/design-system';
+import LoadingHeartBeat from '../../../components/ui/loading-heartbeat';
+
+import TransactionDetailItem from '../../../components/app/transaction-detail-item/transaction-detail-item.component';
+import { NETWORK_TO_NAME_MAP } from '../../../../shared/constants/network';
+import TransactionDetail from '../../../components/app/transaction-detail/transaction-detail.component';
+import ActionableMessage from '../../../components/ui/actionable-message/actionable-message';
+
+const renderHeartBeatIfNotInTest = () =>
+  process.env.IN_TEST ? null : <LoadingHeartBeat />;
+export default class GasDisplay extends Component {
```

Comment:
[REQ] This should be refactored to be a functional component

Reply:
Also drop the .component part of the filename, please :)

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/ce133455d78ea7c4671ac747b61bc6f926d76360).